src/Microsoft.ML.TimeSeries/STL/FastLoess.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// this is the fast version of Loess. there are several alternatives to improve the performance. this one is an approximation approach.
+    /// the smoothing is conducted on a sample set, and then the values on the left points are assigned directly.
+    /// </summary>
+    public class FastLoess
+    {
+        /// <summary>
+        /// this class is a sampling based method, so here specifies the sample size.
+        /// </summary>
+        private static int _sampleSize = 100;
```

Comment:
I'd recommend to change this to `private const int` since it's actually never modified.

Reply:
updated.

---
In reply to: [440499460](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440499460) [](ancestors = 440499460)